deb/ubuntu-cosmic/Dockerfile.x86_64

```diff
@@ -0,0 +1,47 @@
+FROM ubuntu:cosmic
+
+ENV DISTRO ubuntu
+ENV SUITE cosmic
+
+ENV GOPATH /go
+ENV PATH $PATH:/usr/local/go/bin:$GOPATH/bin
+ENV DOCKER_BUILDTAGS apparmor seccomp selinux
+ENV RUNC_BUILDTAGS apparmor seccomp selinux
+
+RUN apt-get update && apt-get install -y \
+	apparmor \
+	bash-completion \
+	libbtrfs-dev \
+	build-essential \
+	cmake \
+	curl \
+	ca-certificates \
+	debhelper \
+	dh-apparmor \
+	dh-systemd \
+	git \
+	libapparmor-dev \
+	libdevmapper-dev \
+	libltdl-dev \
+	libseccomp-dev \
+	pkg-config \
+	libsystemd-dev \
+	--no-install-recommends && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir -p /go/src/github.com/docker && \
+	mkdir -p /go/src/github.com/opencontainers && \
+	mkdir -p /root/build-deb && \
+	ln -snf /engine /root/build-deb/engine && \
+	ln -snf /cli /root/build-deb/cli && \
+	ln -snf /root/build-deb/engine /go/src/github.com/docker/docker && \
+	ln -snf /root/build-deb/cli /go/src/github.com/docker/cli
+
+WORKDIR /root/build-deb
+
+ARG GO_VERSION
+RUN curl -fSL "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" | tar xzC /usr/local
```

Comment:
Is this the only line that differs among each of the 4 Dockerfiles? 
Would it be desirable to have an arch tag as a build arg, and then pass the arch tag in here in a similar way to how the GO_VERSION is being passed in?

That way we don't need to check in and maintain 4 distinct Dockerfiles?

This would probably require a bigger refactor so that the other Dockerfiles for other ubuntu distributions worked in the same way. As a result, if this change is desirable we can track it somewhere.

Reply:
It is the only line that differs and yes it'd require a pretty decent refactor. Let's leave that for another PR.